File: frontend/src/services/petService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api';

const petService = {
  getAllPets: async (category) => {
    const params = category ? { category } : {};
    const response = await axios.get(`${API_BASE_URL}/pets`, { params });
    return response.data;
  },

  getPetById: async (id) => {
    const response = await axios.get(`${API_BASE_URL}/pets/${id}`);
    return response.data;
  },

  createPet: async (pet) => {
    const response = await axios.post(`${API_BASE_URL}/pets`, pet);
    return response.data;
  },

  updatePet: async (id, pet) => {
    const response = await axios.put(`${API_BASE_URL}/pets/${id}`, pet);
    return response.data;
  },

  deletePet: async (id) => {
    await axios.delete(`${API_BASE_URL}/pets/${id}`);
  }
};

export default petService;

File: backend/src/main/java/com/petstore/services/PetService.java
package com.petstore.services;

import com.petstore.models.Pet;
import com.petstore.models.PetCategory;
import com.petstore.repositories.PetRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Service
public class PetService {

    private final PetRepository petRepository;

    @Autowired
    public PetService(PetRepository petRepository) {
        this.petRepository = petRepository;
    }

    public List<Pet> getAllPets() {
        return petRepository.findAll();
    }

    public List<Pet> getPetsByCategory(PetCategory category) {
        return petRepository.findByCategory(category);
    }

    public Optional<Pet> getPetById(UUID id) {
        return petRepository.findById(id);
    }

    public Pet createPet(Pet pet) {
        return petRepository.save(pet);
    }

    public Optional<Pet> updatePet(UUID id, Pet petDetails) {
        return petRepository.findById(id).map(existingPet -> {
            existingPet.setName(petDetails.getName());
            existingPet.setCategory(petDetails.getCategory());
            existingPet.setBreed(petDetails.getBreed());
            existingPet.setAge(petDetails.getAge());
            existingPet.setPrice(petDetails.getPrice());
            existingPet.setImageUrl(petDetails.getImageUrl());
            existingPet.setDescription(petDetails.getDescription());
            return petRepository.save(existingPet);
        });
    }

    public boolean deletePet(UUID id) {
        return petRepository.findById(id).map(pet -> {
            petRepository.delete(pet);
            return true;
        }).orElse(false);
    }
}
